### PR TITLE
New curation-audit command

### PR DESCRIPTION
**Pull request by asafambar**
- [ ] All [tests](https://github.com/jfrog/jfrog-cli#tests) passed. If this feature is not already covered by the tests, I added new tests.
- [ ] All [static analysis checks](https://github.com/jfrog/jfrog-cli/actions/workflows/analysis.yml) passed.
- [ ] This pull request is on the dev branch.
- [ ] I used gofmt for formatting the code before submitting the pull request.
-----

### CODE REVIEW

**Comment by asafambar** [comment]
I have read the CLA Document and I hereby sign the CLA

File: scan/cli.go [hunk 1]
```diff
@@ -33,8 +36,21 @@ import (
 
 const auditScanCategory = "Audit & Scan"
 
+const curationAuditCategory = "Curation Audit"
```
Comment (eyalbe4) [inline scan/cli.go:39]: Let's remove this const and use `auditScanCategory` for the `curation-audit` command category.

File: scan/cli.go [hunk 2]
```diff
@@ -33,8 +36,21 @@ import (
...
+
 func GetCommands() []cli.Command {
 	return cliutils.GetSortedCommands(cli.CommandsByName{
+		{
+			Name:         "curation-audit",
+			Category:     curationAuditCategory,
+			Flags:        cliutils.GetCommandFlags(cliutils.Curation),
+			Aliases:      []string{"caud"},
```
Comment (eyalbe4) [inline scan/cli.go:47]: caud --> ca

File: scan/cli.go [hunk 3]
```diff
@@ -194,6 +210,14 @@ func AuditSpecificCmd(c *cli.Context, technology coreutils.Technology) error {
 	return progressbar.ExecWithProgress(auditCmd)
 }
 
+func CurationCommand(c *cli.Context) error {
```
Comment (eyalbe4) [inline scan/cli.go:213]: CurationCommand --> CurationAuditCmd

File: scan/cli.go [hunk 4]
```diff
@@ -230,13 +252,38 @@ func createGenericAuditCmd(c *cli.Context) (*audit.GenericAuditCommand, error) {
 	if c.String("working-dirs") != "" {
 		auditCmd.SetWorkingDirs(splitAndTrim(c.String("working-dirs"), ","))
 	}
+	auditCmd.GraphBasicParams = &xrCmdUtils.GraphBasicParams{}
```
Comment (eyalbe4) [inline scan/cli.go:255]: It's better to initialize this struct inside the `NewGenericAuditCommand()` function.

File: scan/cli.go [hunk 5]
```diff
@@ -230,13 +252,38 @@ func createGenericAuditCmd(c *cli.Context) (*audit.GenericAuditCommand, error) {
...
+	return auditCmd, err
+}
 
-	return auditCmd.SetExcludeTestDependencies(c.Bool(cliutils.ExcludeTestDeps)).
-			SetUseWrapper(c.BoolT(cliutils.UseWrapper)).
-			SetInsecureTls(c.Bool(cliutils.InsecureTls)).
-			SetNpmScope(c.String(cliutils.DepType)).
-			SetPipRequirementsFile(c.String(cliutils.RequirementsFile)),
-		err
+func createCurtainCmd(c *cli.Context) (*curation.Command, error) {
```
Comment (eyalbe4) [inline scan/cli.go:266]: This function seems redundant. Let's move its content into the `CurationAuditCmd` function.

File: scan/cli.go (hunk 3 above)
Comment (eyalbe4) [inline scan/cli.go:213]: CurationCommand  --> CurationAuditCmd

File: utils/cliutils/commandsflags.go [hunk 6]
```diff
@@ -103,6 +103,7 @@ const (
 
 	// Xray's Commands Keys
 	XrCurl        = "xr-curl"
+	Curation      = "curation-audit"
```
Comment (eyalbe4) [inline utils/cliutils/commandsflags.go:106]: Curation --> CurationAudit

File: utils/cliutils/commandsflags.go [hunk 7]
```diff
@@ -1818,6 +1819,9 @@ var commandFlags = map[string][]string{
 	XrCurl: {
 		serverId,
 	},
+	Curation: {
+		xrOutput, DepType, workingDirs,
```
Comment (eyalbe4) [inline utils/cliutils/commandsflags.go:1823]: The `DepType` is currently redundant and should be removed.

**Review by eyalbe4 — CHANGES_REQUESTED** [review]
Is `sarif` supported as an output format? If not, let's set the correct option description for this command.
image

Let's document the new command in the `CLI-for-JFrog-Xray.md` file.

File: scan/cli.go [hunk 8]
```diff
@@ -193,6 +207,32 @@ func AuditSpecificCmd(c *cli.Context, technology coreutils.Technology) error {
 	auditCmd.SetTechnologies(technologies)
 	return progressbar.ExecWithProgress(auditCmd)
 }
+func CurationCmd(c *cli.Context) error {
```
Comment (eyalbe4) [inline scan/cli.go:210]: Missing new-line before function.

File: xray_test.go [hunk 9]
```diff
@@ -608,3 +617,117 @@ func TestXrayOfflineDBSyncV3(t *testing.T) {
...
+
+func TestCurationAudit(t *testing.T) {
+	initXrayTest(t, "")
+	tempDirPath, createTempDirCallback := coreTests.CreateTempDirWithCallbackAndAssert(t)
+	defer createTempDirCallback()
+	multiProject := filepath.Join(filepath.FromSlash(tests.GetTestResourcesPath()), "xray")
+	assert.NoError(t, fileutils.CopyDir(multiProject, tempDirPath, true, nil))
+	rootDir, err := os.Getwd()
+	require.NoError(t, err)
+	defer os.Chdir(rootDir)
```
Comment (eyalbe4) [inline xray_test.go:629]: Should be replaced with - 
```
	defer func() {
		assert.NoError(t, os.Chdir(rootDir))
	}()
```

**Review by eyalbe4 — CHANGES_REQUESTED** [review]

**Pull request merged**